test/board_test.rb

```diff
@@ -0,0 +1,201 @@
+
+  def test_it_can_identify_whether_coordinates_are_consecutive
+    assert_equal true, @board.coordinates_consecutive?(["A1", "A2", "A3"])
+    assert_equal true, @board.coordinates_consecutive?(["A1", "B1", "C1"])
+    assert_equal false, @board.coordinates_consecutive?(["A1", "A2", "A4"])
+    assert_equal false, @board.coordinates_consecutive?(["A1", "C1"])
+    assert_equal false, @board.coordinates_consecutive?(["A3", "A2", "A1"])
+    assert_equal false, @board.coordinates_consecutive?(["C1", "B1"])
+  end
+
+  def test_coordinates_must_be_same_length_as_ship
+    assert_equal false, @board.valid_placement?(@cruiser, ["A1", "A2"])
+    assert_equal false, @board.valid_placement?(@submarine, ["A2", "A3", "A4"])
+
+    assert_equal true, @board.valid_placement?(@cruiser, ["A1", "A2", "A3"])
+    assert_equal true, @board.valid_placement?(@submarine, ["A2", "A3"])
+  end
+
+  def test_coordinates_must_be_consecutive
+    assert_equal false, @board.valid_placement?(@cruiser, ["A1", "A2", "A4"])
+    assert_equal false, @board.valid_placement?(@submarine, ["A1", "C1"])
+    assert_equal false, @board.valid_placement?(@cruiser, ["A3", "A2", "A1"])
+    assert_equal false, @board.valid_placement?(@submarine, ["C1", "B1"])
+
+    assert_equal true, @board.valid_placement?(@cruiser, ["A1", "A2", "A3"])
+    assert_equal true, @board.valid_placement?(@cruiser, ["A1", "B1", "C1"])
+    assert_equal true, @board.valid_placement?(@submarine, ["C1", "D1"])
+    assert_equal true, @board.valid_placement?(@submarine, ["D1", "D2"])
+  end
+
+  def test_coordinates_cant_be_diagonal
+
+    assert_equal false, @board.valid_placement?(@cruiser, ["A1", "B2", "C3"])
+    assert_equal false, @board.valid_placement?(@submarine, ["C2", "D3"])
+  end
+
+  def test_it_can_use_all_criteria_to_validate_placement
+    assert_equal true, @board.valid_placement?(@submarine, ["A1", "A2"])
+    assert_equal true, @board.valid_placement?(@cruiser, ["B1", "C1", "D1"])
+  end
+
+  def test_ships_occupy_multiple_cells
+    @board.place(@cruiser, ["A1", "A2", "A3"])
+    cell_1 = @board.cells["A1"]
+    cell_2 = @board.cells["A2"]
+    cell_3 = @board.cells["A3"]
+
+    assert_equal @cruiser, cell_1.ship
+    assert_equal @cruiser, cell_2.ship
+    assert_equal @cruiser, cell_3.ship
+  end
+
+  def test_if_all_cells_are_empty
+    @board.place(@cruiser, ["A1", "A2", "A3"])
+    assert_equal true, @board.cell_has_no_ships?(["B2", "B3", "B4"])
+    assert_equal false, @board.cell_has_no_ships?(["A1","A2","A3"])
+  end
+
+  def test_ships_do_not_overlap
+    cell_1 = @board.cells["A1"]
+    cell_2 = @board.cells["A2"]
+    cell_3 = @board.cells["A3"]
+    @board.place(@cruiser, ["A1", "A2", "A3"])
+    @board.place(@submarine, ["A1", "A2"])
+
+    assert_equal @cruiser, cell_1.ship
+    assert_equal @cruiser, cell_2.ship
+    assert_equal @cruiser, cell_3.ship
+    assert_equal false, @board.valid_placement?(@submarine, ["A1", "B1"])
+  end
+
+  def test_it_can_render_default
```

Comment:
These are a great start for testing the rendering method! And it's awesome that you're using different test methods to confirm what a rendered board looks like in different states. It'd probably be good to get closer to testing the structure of the board itself though. Since we can't directly test the full board as the expected value argument within an assertion, we could define an `empty_board` variable to equal
```ruby
"  1 2 3 4 \n" +
"A . . . . \n" +
"B . . . . \n" +
"C . . . . \n" +
"D . . . . \n"
```
And then use `assert_equal empty_board, @board.render`